deps/linenoise/linenoise.c

```diff
@@ -103,7 +103,7 @@
  *
  */
 
-#define _BSD_SOURCE     /* For fchmod() */
+#define _DEFAULT_SOURCE /* For fchmod() */
```

Comment:
i see that in `src/fmacros.h` and `tests/modules/fork.c` we have both, maybe that's what we need to do?

Reply:
This is from the man page:

Since glibc 2.20, this macro is deprecated.  It now has the same effect as defining _DEFAULT_SOURCE, but generates a compile-time warning (unless _DEFAULT_SOURCE is also defined).  Use  _DE‐
               FAULT_SOURCE  instead.  To allow code that requires _BSD_SOURCE in glibc 2.19 and earlier and _DEFAULT_SOURCE in glibc 2.20 and later to compile without warnings, define both _BSD_SOURCE and
               _DEFAULT_SOURCE.

So it makes sense to have both only if we expect to be built with glibc 2.19 or earlier. 2.20 was released 7 years ago, so I think we can safely say the latest Redis doesn't support glibc 2.19. But if you insist we can have both.

Reply:
what about other compilers? or maybe old embedded toolchains?
what do we have to lose by adding both? (nothing).

the way i look at it, if there's some practice in one part of redis that proves to be "stable" we can mirror that practice to the other part.